src/test/java/io/debezium/connector/spanner/config/validation/ConnectionValidatorTest.java

```diff
@@ -7,51 +7,99 @@
+        Map<String, ConfigValue> configValueMap = Map.of(
+                "gcp.spanner.project.id", new ConfigValue("gcp.spanner.project.id", "boxwood-weaver-353315", new ArrayList<>(), new ArrayList<>()),
+                "gcp.spanner.instance.id", new ConfigValue("gcp.spanner.instance.id", "kafka-connector", new ArrayList<>(), new ArrayList<>()),
+                "gcp.spanner.database.id", new ConfigValue("gcp.spanner.database.id", "kafkaspan", new ArrayList<>(), new ArrayList<>()),
+                "gcp.spanner.credentials.path", new ConfigValue("gcp.spanner.credentials.json", "{}", new ArrayList<>(), new ArrayList<>()),
+                "gcp.spanner.credentials.json", new ConfigValue("gcp.spanner.credentials.json", "{}", new ArrayList<>(), new ArrayList<>()),
+                "gcp.spanner.host", new ConfigValue("gcp.spanner.host", "http://localhost:9010", new ArrayList<>(), new ArrayList<>()),
+                "gcp.spanner.emulator.host", new ConfigValue("gcp.spanner.emulator.host", "http://localhost:9010", new ArrayList<>(), new ArrayList<>()));
+
+        ConfigurationValidator.ValidationContext validationContext = new ConfigurationValidator.ValidationContext(configuration, configValueMap);
+
+        ConnectionValidator connectionValidator = spy(ConnectionValidator.withContext(validationContext));
+        connectionValidator.validate();
+        Assertions.assertEquals(false, connectionValidator.isSuccess());
+    }
+
+    @Test
+    void validateFailForMissingProperties() {
+        Configuration configuration = Configuration.from(Map.of(
+                "gcp.spanner.credentials.path", "no_path",
+                "gcp.spanner.database.id", "kafkaspan"));
+        List<String> errMessages = List.of("missing required property");
+        Map<String, ConfigValue> configValueMap = Map.of(
+                "gcp.spanner.project.id", new ConfigValue("gcp.spanner.project.id", "boxwood-weaver-353315", new ArrayList<>(), errMessages),
+                "gcp.spanner.instance.id", new ConfigValue("gcp.spanner.instance.id", "kafka-connector", new ArrayList<>(), errMessages),
+                "gcp.spanner.database.id", new ConfigValue("gcp.spanner.database.id", "kafkaspan", new ArrayList<>(), new ArrayList<>()),
+                "gcp.spanner.credentials.path", new ConfigValue("gcp.spanner.credentials.path", "no_path", new ArrayList<>(), new ArrayList<>()),
+                "gcp.spanner.credentials.json", new ConfigValue("gcp.spanner.credentials.json", "{}", new ArrayList<>(), new ArrayList<>()));
+
+        ConfigurationValidator.ValidationContext validationContext = new ConfigurationValidator.ValidationContext(configuration, configValueMap);
+        ConnectionValidator connectionValidator = spy(ConnectionValidator.withContext(validationContext));
         connectionValidator.validate();
-        Assertions.assertEquals(isSuccess, connectionValidator.isSuccess());
+        Assertions.assertEquals(false, connectionValidator.isSuccess());
```

Comment:
Which property is missing here?

Reply:
I have updated this test and it should be more clear now. As you can see the configuration only contain `gcp.spanner.database.id`, which misses other project/instance ids and the credentials.